src/api.c

```diff
@@ -731,3 +731,28 @@ API int seccomp_export_bpf(const scmp_filter_ctx ctx, int fd)
 
 	return 0;
 }
+
+/* NOTE - function header comment in include/seccomp.h */
+API int seccomp_export_bpf_mem(const scmp_filter_ctx ctx, void *buf,
+			       size_t *len)
+{
+	size_t buf_len;
+	struct bpf_program *program;
+
+	if (_ctx_valid(ctx) || !buf || !len)
+		return -EINVAL;
+
+	program = gen_bpf_generate((struct db_filter_col *)ctx);
```

Comment:
Note that `gen_bpf_generate()` was changed in commit 3a1d1c977065f.  It now returns an `rc`, and takes two parameters - `struct db_filter_col *` and `struct bpf_program *`.

Reply:
updated